filament/backend/src/BackendUtils.cpp

```diff
@@ -16,9 +16,53 @@
 
 #include "private/backend/BackendUtils.h"
 
+#include <string_view>
+
 namespace filament {
 namespace backend {
 
+bool requestsGoogleLineDirectivesExtension(const char* shader, size_t length) noexcept {
+    std::string_view s(shader, length);
```

Comment:
Could we just assume a shorter length? i.e. the directive will be in the first 1 KiB or less? It seems like there's no reason to parse the entire shader when the directive is not present (is it ever not present though? )

Reply:
It's not present in optimized GLSL. Yeah, I can limit this to the first 1 KiB.